sql/src/main/java/io/crate/metadata/FunctionInfo.java

```diff
@@ -227,6 +227,7 @@ public void writeTo(StreamOutput out) throws IOException {
          * So for f comparisons cannot be replaced.
          */
         COMPARISON_REPLACEMENT,
-        LAZY_ATTRIBUTES
+        LAZY_ATTRIBUTES,
+        CONDITIONAL
```

Comment:
I didn't think about this earlier - but this afaik breaks the rolling upgrade functionality due to a streaming change? (E.g. if the handler node already sets this attribute, but it cannot be read on a collect node running on the old version or something like that)

If this is supposed to go into 2.2., I guess a static list would be required after all?

Also, I'm not sure if `CONDITIONAL` conveys the right aspect. After all any function that takes boolean arguments might treat `null` values different than `false`. So I guess a name that conveys that this functions needs to distinguish between false/null would fit better.